parsec/core/fs/workspacefs/file_transactions.py

```diff
@@ -233,4 +235,5 @@ def _attempt_read(self, manifest: LocalFileManifest, start: int, end: int):
 
     async def fd_flush(self, fd: FileDescriptor) -> None:
         # No-op
-        pass
+        async with self._load_and_lock_file(fd) as (entry_id, manifest):
+            self.local_storage.set_manifest(entry_id, manifest)
```

Comment:
This should do the trick :+1:

Reply:
I've improved this to avoid doing the costly serialization if the manifest's cache wasn't ahead of the disk